lib/ansible/modules/remote_management/ucs/ucs_dns_server.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: ucs_dns_server
+short_description: Configures DNS server on Cisco UCS Manager
+extends_documentation_fragment:
+- ucs
+description:
+- Configures DNS server on Cisco UCS Manager.
+- Examples can be used with the L(UCS Platform Emulator,https://communities.cisco.com/ucspe).
+options:
+  state:
+    description:
+    - If C(absent), will remove a DNS server.
+    - If C(present), will add or update a DNS server.
+    choices: [absent, present]
+    default: present
+
+  dns_server:
+    description:
+    - DNS server IP address.
+    - Enter a valid IPV4 Address.
+    aliases: [ name ]
+    default: ""
+
+  description:
+    description:
+    - A user-defined description of the DNS server.
+    - Enter up to 256 characters.
+    - "You can use any characters or spaces except the following:"
+    - "` (accent mark), \ (backslash), ^ (carat), \" (double quote), = (equal sign), > (greater than), < (less than), or ' (single quote)."
+    aliases: [ descr ]
+    default: ""
+
+requirements:
+- ucsmsdk
+author:
+- John McDonough (@movinalot)
+- CiscoUcs (@CiscoUcs)
+version_added: "2.7"
+'''
+
+EXAMPLES = r'''
+- name: Configure DNS server
+  ucs_dns_server:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    dns_server: 10.10.10.10
+    description: DNS Server IP address
+    state: present
```

Comment:
So we have had some discussions about modules that are meant to run on the Ansible controller and connect to a remote device. I personally think that adding `delegate_to: localhost` in the examples makes it very clear to the user that this is not a module you run on the target.

That's why we started to add this to other modules as well. So the examples work without having to do anything special in the background.

Reply:
I have add delegate_to: localhost